### PR TITLE
/!\ WIP /!\ Specs for security draft mode

### PR DESCRIPTION
**Pull request by Doude**

### CODE REVIEW

File: security_draft_mode.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+1. The modification create a new security resource
+
+   In that case, we create that new security resource with a draft flag set to
+   True. All reference are created and identical to a normal resource.
+
+   Followed modification on that created security resource will be append to it.
+
+2. The modification update an exiting security resource
+
+   Here, we clone the security resource, set the draft flag, reference it with
```
Comment (Doude) [inline security_draft_mode.md]: How we clone UUID and fq_name?

Comment (sbansal7) [reply security_draft_mode.md]: The clone objects will have different uuid/fq-name. fq-name could be derived from the main object

Comment (Doude) [reply security_draft_mode.md]: Ok for the fq_name, but isn't it weird for the user if the UUID change? And it's probably going to cause problems to automation tools loke Heat or terraform, no?

File: security_draft_mode.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+After, the security team/person, can review that proposed security modification
+thanks to new REST API calls to list and show them. Security team/person have
+two choices:
+
+1. Commit modifications
+
+   When security resource modifications were validated and ready to be enforced,
+   the security team/person can commit them. In that first version, the commit
+   granularity will be limited to the scope. When a commit is on going, no more
+   security modifications are allowed.
```
Comment (sbansal7) [inline security_draft_mode.md]: Please describe what operations will be performed by API server when commit is requested. I.e., for every object with a draft copy, we will delete the contents of the existing row in cassandra, replace with contents of the draft row (keeping uuid and fq_name same as before) and then delete the draft row.

For objects with 'deleted' flag, we will delete both the original as well as draft objects. For 'new' objects, we will simply reset the 'draft' flag.

File: security_draft_mode.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+2. Abandon modifications
+
+   When security resource modifications were not validated, the security
+   team/person can abandon them. In that first version, the abandon/revert
+   granularity will be limited to the scope. When an abandon/revert is on going,
+   no more security modifications are allowed.
```
Comment (sbansal7) [inline security_draft_mode.md]: Please describe what actions will be performed by api server when abandon is requested. I.e., all objects with draft flag will be deleted.

File: security_draft_mode.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+2. Add `draft` flag to the five security resources
+
+   That flag permits to know which security resources were created/modified for
+   for audit and commit.
+
+3. Add `to_delete` flag to the five security resources
+
+   That flag determines which security resources were delete for audit and
+   commit.
+
```
Comment (sbansal7) [inline security_draft_mode.md]: We will also add a link from each of the 5 security objects to a draft object of the same type.

File: security_draft_mode.md (hunk 2 above)
Comment (Doude) [reply security_draft_mode.md]: Ok I added that in §3.2

File: security_draft_mode.md (hunk 3 above)
Comment (Doude) [reply security_draft_mode.md]: Ok I added that in §3.2

File: security_draft_mode.md (hunk 1 above)
Comment (Doude) [reply security_draft_mode.md]: I get my answer from your comment above

**Pull request closed without merging**